<a href="https://colab.research.google.com/github/Suhail-46/Analysis-of-datasets-in-Nutrition-Paradox/blob/main/DS_Nutrition_Paradox_A_Global_View_on_Obesity%26Malnutrition.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Code written to send request to API and scrape data in form of JSON(return data in dictionary format).
import requests
import pprint
url_obesity_adults = "https://ghoapi.azureedge.net/api/NCD_BMI_30C"
url_obesity_children = "https://ghoapi.azureedge.net/api/NCD_BMI_PLUS2C"
url_underweight_adults = "https://ghoapi.azureedge.net/api/NCD_BMI_18C"
url_underweight_children = "https://ghoapi.azureedge.net/api/NCD_BMI_MINUS2C"
response_obesity_adults = requests.get(url_obesity_adults)
response_obesity_children = requests.get(url_obesity_children)
response_underweight_adults = requests.get(url_underweight_adults)
response_underweight_children = requests.get(url_underweight_children)
data_obesity_adults = response_obesity_adults.json()
data_obesity_children = response_obesity_children.json()
data_underweight_adults = response_underweight_adults.json()
data_underweight_children = response_underweight_children.json()

In [2]:
# Code written to convert the data from dictionary to dataframes using pandas
import pandas as pd
df1 = pd.DataFrame(data_obesity_adults['value'])
df2 = pd.DataFrame(data_obesity_children['value'])
df3 = pd.DataFrame(data_underweight_adults['value'])
df4 = pd.DataFrame(data_underweight_children['value'])

In [3]:
# Validating with single dataframe by printing output.
df1.head()

Id IndicatorCode SpatialDimType SpatialDim ParentLocationCode  \
0  5395519   NCD_BMI_30C        COUNTRY        SAU                EMR   
1  5395636   NCD_BMI_30C        COUNTRY        LKA               SEAR   
2  5395898   NCD_BMI_30C        COUNTRY        KAZ                EUR   
3      261   NCD_BMI_30C        COUNTRY        KHM                WPR   
4      358   NCD_BMI_30C        COUNTRY        IRN                EMR   

  TimeDimType         ParentLocation Dim1Type  TimeDim      Dim1  ...  \
0        YEAR  Eastern Mediterranean      SEX     1997  SEX_BTSX  ...   
1        YEAR        South-East Asia      SEX     2013   SEX_MLE  ...   
2        YEAR                 Europe      SEX     2005  SEX_FMLE  ...   
3        YEAR        Western Pacific      SEX     2001  SEX_BTSX  ...   
4        YEAR  Eastern Mediterranean      SEX     2020   SEX_MLE  ...   

  DataSourceDim             Value NumericValue        Low       High Comments  \
0          None  22.1 [21.1-23.2]    22.111502  21.063206  23.219305     None   
1          None     3.7 [3.0-4.4]     3.682037   3.004463   4.399506     None   
2          None  19.0 [16.9-21.3]    19.026031  16.881686  21.300718     None   
3          None     1.0 [0.8-1.3]     0.981696   0.756741   1.274868     None   
4          None  17.5 [15.8-19.2]    17.487641  15.793777  19.245727     None   

                            Date  TimeDimensionValue  \
0  2024-02-29T16:06:41.017+01:00                1997   
1  2024-02-29T16:06:41.017+01:00                2013   
2  2024-02-29T16:06:41.017+01:00                2005   
3  2024-02-29T16:06:41.017+01:00                2001   
4  2024-02-29T16:06:41.017+01:00                2020   

          TimeDimensionBegin           TimeDimensionEnd  
0  1997-01-01T00:00:00+01:00  1997-12-31T00:00:00+01:00  
1  2013-01-01T00:00:00+01:00  2013-12-31T00:00:00+01:00  
2  2005-01-01T00:00:00+01:00  2005-12-31T00:00:00+01:00  
3  2001-01-01T00:00:00+01:00  2001-12-31T00:00:00+01:00  
4  2020-01-01T00:00:00+01:00  2020-12-31T00:00:00+01:00  

[5 rows x 25 columns]

In [4]:
# Code written to add new column for each dataframe along with adding values to the column
df1['Age_group'] = 'adults'
df2['Age_group'] = 'children'
df3['Age_group'] = 'adults'
df4['Age_group'] = 'children'

In [5]:
# Validating with single dataframe by printing columns.
df1.columns

Index(['Id', 'IndicatorCode', 'SpatialDimType', 'SpatialDim',
       'ParentLocationCode', 'TimeDimType', 'ParentLocation', 'Dim1Type',
       'TimeDim', 'Dim1', 'Dim2Type', 'Dim2', 'Dim3Type', 'Dim3',
       'DataSourceDimType', 'DataSourceDim', 'Value', 'NumericValue', 'Low',
       'High', 'Comments', 'Date', 'TimeDimensionValue', 'TimeDimensionBegin',
       'TimeDimensionEnd', 'Age_group'],
      dtype='object')

In [6]:
# Combined the two obesity dataframes into one dataframe called df_obesity
df_obesity = pd.concat([df1, df2], axis=0)
# Combined the two underweight dataframes into one dataframe called df_malnutrition
df_malnutrition = pd.concat([df3, df4], axis=0)

In [7]:
# Filtered dataframes which includes only records from the years 2012 to 2022
df_obesity_filtered = df_obesity[(df_obesity['TimeDim'] >= 2012) & (df_obesity['TimeDim'] <= 2022)]
df_malnutrition_filtered = df_malnutrition[(df_malnutrition['TimeDim'] >= 2012) & (df_malnutrition['TimeDim'] <= 2022)]

In [8]:
# Extracted only specific columns from each dataframe as per requirement of project
df_obesity_clean = df_obesity_filtered[['ParentLocation','Dim1', 'TimeDim', 'Low', 'High', 'NumericValue', 'SpatialDim', 'Age_group']]
df_malnutrition_clean = df_malnutrition_filtered[['ParentLocation','Dim1', 'TimeDim', 'Low', 'High', 'NumericValue', 'SpatialDim', 'Age_group']]

In [9]:
# Validation is made by printing output of df_obesity_clean dataframe to ensure that required columns filtered from df_obesity_filtered dataframe.
df_obesity_clean.head()

ParentLocation      Dim1  TimeDim        Low       High  \
1         South-East Asia   SEX_MLE     2013   3.004463   4.399506   
4   Eastern Mediterranean   SEX_MLE     2020  15.793777  19.245727   
11  Eastern Mediterranean   SEX_MLE     2020  23.089589  33.815336   
17                 Europe  SEX_FMLE     2014  24.924766  31.213260   
19                 Europe  SEX_FMLE     2019  28.160682  35.574925   

    NumericValue SpatialDim Age_group  
1       3.682037        LKA    adults  
4      17.487641        IRN    adults  
11     28.335530        IRQ    adults  
17     27.948704        GRL    adults  
19     31.893221        ARM    adults

In [10]:
# Code written to rename the columns into meaningful names.
df_obesity_clean.rename(columns={'ParentLocation': 'Region', 'Dim1': 'Gender', 'TimeDim': 'Year', 'Low': 'LowerBound',
                                 'High': 'UpperBound', 'NumericValue': 'Mean_Estimate', 'SpatialDim': 'Country'}, inplace=True)
df_malnutrition_clean.rename(columns={'ParentLocation': 'Region', 'Dim1': 'Gender', 'TimeDim': 'Year', 'Low': 'LowerBound',
                                 'High': 'UpperBound', 'NumericValue': 'Mean_Estimate', 'SpatialDim': 'Country'}, inplace=True)

/tmp/ipython-input-10-1276501761.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_obesity_clean.rename(columns={'ParentLocation': 'Region', 'Dim1': 'Gender', 'TimeDim': 'Year', 'Low': 'LowerBound',
/tmp/ipython-input-10-1276501761.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_malnutrition_clean.rename(columns={'ParentLocation': 'Region', 'Dim1': 'Gender', 'TimeDim': 'Year', 'Low': 'LowerBound',


In [11]:
#  Validation is made by printing output of df_obesity_clean dataframe to ensure that columns are renamed
df_obesity_clean

Region    Gender  Year  LowerBound  UpperBound  \
1            South-East Asia   SEX_MLE  2013    3.004463    4.399506   
4      Eastern Mediterranean   SEX_MLE  2020   15.793777   19.245727   
11     Eastern Mediterranean   SEX_MLE  2020   23.089589   33.815336   
17                    Europe  SEX_FMLE  2014   24.924766   31.213260   
19                    Europe  SEX_FMLE  2019   28.160682   35.574925   
...                      ...       ...   ...         ...         ...   
62347                 Europe  SEX_BTSX  2020    0.964176    2.197432   
62350  Eastern Mediterranean  SEX_FMLE  2012    0.302039    5.872414   
62358        Western Pacific   SEX_MLE  2021    3.325899   34.877816   
62361                 Africa   SEX_MLE  2014    0.236481    6.341185   
62364                 Europe   SEX_MLE  2017   10.087850   12.760153   

       Mean_Estimate Country Age_group  
1           3.682037     LKA    adults  
4          17.487641     IRN    adults  
11         28.335530     IRQ    adults  
17         27.948704     GRL    adults  
19         31.893221     ARM    adults  
...              ...     ...       ...  
62347       1.475931     TJK  children  
62350       1.966190     SOM  children  
62358      16.634887     VUT  children  
62361       2.032189     MDG  children  
62364      11.390631     FIN  children  

[27720 rows x 8 columns]

In [12]:
# Checking the unique values of Gender column to change the values into meaningful categorical values
df_obesity_clean["Gender"].unique()

array(['SEX_MLE', 'SEX_FMLE', 'SEX_BTSX'], dtype=object)

In [13]:
#  Code written to change the values of Gender column into meaningful categorical values.
df_obesity_clean['Gender'] = df_obesity_clean['Gender'].replace({'SEX_MLE': 'Male', 'SEX_FMLE': 'Female', 'SEX_BTSX': 'Both'})
df_malnutrition_clean['Gender'] = df_malnutrition_clean['Gender'].replace({'SEX_MLE': 'Male', 'SEX_FMLE': 'Female', 'SEX_BTSX': 'Both'})

/tmp/ipython-input-13-2330079318.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_obesity_clean['Gender'] = df_obesity_clean['Gender'].replace({'SEX_MLE': 'Male', 'SEX_FMLE': 'Female', 'SEX_BTSX': 'Both'})
/tmp/ipython-input-13-2330079318.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_malnutrition_clean['Gender'] = df_malnutrition_clean['Gender'].replace({'SEX_MLE': 'Male', 'SEX_FMLE': 'Female', 'SEX_BTSX': 'Both'})


In [14]:
# Validating  with df_obesity_clean dataframe by printing first five columns to check that values changed
df_obesity_clean.head()

Region  Gender  Year  LowerBound  UpperBound  \
1         South-East Asia    Male  2013    3.004463    4.399506   
4   Eastern Mediterranean    Male  2020   15.793777   19.245727   
11  Eastern Mediterranean    Male  2020   23.089589   33.815336   
17                 Europe  Female  2014   24.924766   31.213260   
19                 Europe  Female  2019   28.160682   35.574925   

    Mean_Estimate Country Age_group  
1        3.682037     LKA    adults  
4       17.487641     IRN    adults  
11      28.335530     IRQ    adults  
17      27.948704     GRL    adults  
19      31.893221     ARM    adults

In [15]:
# Checking the unique values of Country column to make validations further.
df_obesity_clean["Country"].nunique()

210

In [16]:
# Installing pycountry package and importing it for convertion of 3-letter country codes (ISO Alpha-3) into full country names in the Country column
!pip install pycountry
import pycountry

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.3/6.3 MB 54.7 MB/s eta 0:00:00


We know the undefined values of pycountry package present in the dataframes from the documentation as special cases so replacing them as per requirement.

In [17]:
special_cases = {
                    'GLOBAL': 'Global',
                    'WB_LMI': 'Low & Middle Income',
                    'WB_HI': 'High Income',
                    'WB_LI': 'Low Income',
                    'EMR': 'Eastern Mediterranean Region',
                    'EUR': 'Europe',
                    'AFR': 'Africa',
                    'SEAR': 'South-East Asia Region',
                    'WPR': 'Western Pacific Region',
                    'AMR': 'Americas Region',
                    'WB_UMI': 'Upper Middle Income'}
df_obesity_clean['Country'] = df_obesity_clean['Country'].replace(special_cases)
df_malnutrition_clean['Country'] = df_malnutrition_clean['Country'].replace(special_cases)

/tmp/ipython-input-17-1303247804.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_obesity_clean['Country'] = df_obesity_clean['Country'].replace(special_cases)
/tmp/ipython-input-17-1303247804.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_malnutrition_clean['Country'] = df_malnutrition_clean['Country'].replace(special_cases)


In [18]:
# Code written to convert 3-letter country codes (ISO Alpha-3) into full country names in the Country column
country_dict = {country.alpha_3: country.name for country in pycountry.countries}
df_obesity_clean['Country'] = df_obesity_clean['Country'].apply(lambda x: country_dict.get(x, x))
df_malnutrition_clean['Country'] = df_malnutrition_clean['Country'].apply(lambda x: country_dict.get(x, x))
# country_dict.get(x, x) method is used to ignore the undefined values of pycountry package

/tmp/ipython-input-18-3584091097.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_obesity_clean['Country'] = df_obesity_clean['Country'].apply(lambda x: country_dict.get(x, x))
/tmp/ipython-input-18-3584091097.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_malnutrition_clean['Country'] = df_malnutrition_clean['Country'].apply(lambda x: country_dict.get(x, x))


In [19]:
# Displaying df_obesity_clean dataframe to verify convertion in Country column
df_obesity_clean.head()

Region  Gender  Year  LowerBound  UpperBound  \
1         South-East Asia    Male  2013    3.004463    4.399506   
4   Eastern Mediterranean    Male  2020   15.793777   19.245727   
11  Eastern Mediterranean    Male  2020   23.089589   33.815336   
17                 Europe  Female  2014   24.924766   31.213260   
19                 Europe  Female  2019   28.160682   35.574925   

    Mean_Estimate                    Country Age_group  
1        3.682037                  Sri Lanka    adults  
4       17.487641  Iran, Islamic Republic of    adults  
11      28.335530                       Iraq    adults  
17      27.948704                  Greenland    adults  
19      31.893221                    Armenia    adults

I have noticed that Region column with None as value is having special cases of Country column

In [20]:
# Validating the replacement of special cases in Country column using region column
df_obesity_clean[df_obesity_clean['Region'].isnull()]

Region  Gender  Year  LowerBound  UpperBound  Mean_Estimate  \
155     None  Female  2016    5.970943    6.696701       6.324915   
167     None  Female  2019   24.963759   26.624760      25.783622   
251     None  Female  2017   12.172082   12.886313      12.520297   
352     None  Female  2022    9.591366   11.653028      10.585164   
685     None    Male  2021   29.186985   32.267982      30.725309   
...      ...     ...   ...         ...         ...            ...   
61963   None    Male  2020    3.208165    6.375460       4.610661   
61999   None    Male  2020    8.139845    9.995961       9.013220   
62017   None    Male  2014    3.995784    5.021876       4.487799   
62044   None    Male  2022    9.036333   14.086503      11.429928   
62303   None    Male  2013   13.517177   14.528246      14.011872   

                            Country Age_group  
155          Western Pacific Region    adults  
167                     High Income    adults  
251             Low & Middle Income    adults  
352          South-East Asia Region    adults  
685                 Americas Region    adults  
...                             ...       ...  
61963                    Low Income  children  
61999                        Europe  children  
62017        South-East Asia Region  children  
62044  Eastern Mediterranean Region  children  
62303                   High Income  children  

[1452 rows x 8 columns]

In [20]:
# Adding CI_Width as new column to dataframe which returns difference of UpperBound and LowerBound as per requirement
df_obesity_clean['CI_Width']  = df_obesity_clean['UpperBound'] - df_obesity_clean['LowerBound']
df_malnutrition_clean['CI_Width']  = df_malnutrition_clean['UpperBound'] - df_malnutrition_clean['LowerBound']

/tmp/ipython-input-20-1185898036.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_obesity_clean['CI_Width']  = df_obesity_clean['UpperBound'] - df_obesity_clean['LowerBound']
/tmp/ipython-input-20-1185898036.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_malnutrition_clean['CI_Width']  = df_malnutrition_clean['UpperBound'] - df_malnutrition_clean['LowerBound']


In [21]:
# Running this code to intake count of rows and columns for validations
df_obesity_clean[df_obesity_clean['Mean_Estimate'] >= 30].shape

(1929, 9)

In [22]:
# Defined a list of conditions and corresponding values. Using np.select() from NumPy, added a new column called obesity_level with the assigned conditions.
import numpy as np
conditions = [
    (df_obesity_clean['Mean_Estimate'] < 25),
    (df_obesity_clean['Mean_Estimate'] >= 25) & (df_obesity_clean['Mean_Estimate'] <= 29.9),
    (df_obesity_clean['Mean_Estimate'] >= 30)
]
choices = ['Low', 'Moderate', 'High']
df_obesity_clean['Obesity_level'] = np.select(conditions, choices, default='Unknown')

/tmp/ipython-input-22-1694476204.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_obesity_clean['Obesity_level'] = np.select(conditions, choices, default='Unknown')


In [23]:
# Validating count of row with code ran for checking Mean_Estimate >= 30
df_obesity_clean[df_obesity_clean['Obesity_level'] == 'High'].shape

(1929, 10)

From the validations, the row and column counts match before and after adding the obesity_level column. Hence, it is verified that the new category column (obesity_level) was added successfully.

In [24]:
# Running this code to intake count of rows and columns for validations
df_malnutrition_clean[df_malnutrition_clean['Mean_Estimate'] >= 20].shape

(502, 9)

In [25]:
# Defined a list of conditions and corresponding values. Using np.select() from NumPy, added a new column called malnutrition_level with the assigned conditions.
import numpy as np
conditions = [
    (df_malnutrition_clean['Mean_Estimate'] < 10),
    (df_malnutrition_clean['Mean_Estimate'] >= 10) & (df_malnutrition_clean['Mean_Estimate'] <= 19.9),
    (df_malnutrition_clean['Mean_Estimate'] >= 20)
]
choices = ['Low', 'Moderate', 'High']
df_malnutrition_clean['Malnutrition_level'] = np.select(conditions, choices, default='Unknown')

/tmp/ipython-input-25-2874632775.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_malnutrition_clean['Malnutrition_level'] = np.select(conditions, choices, default='Unknown')


In [26]:
# Validating count of row with code ran for checking Mean_Estimate >= 20
df_malnutrition_clean[df_malnutrition_clean['Malnutrition_level'] == 'High'].shape

(502, 10)

From the validations, the row and column counts match before and after adding the malnutrition_level column. Hence, it is verified that the new category column (malnutrition_level) was added successfully.

In [27]:
# Hence all requirements are completed so displaying cleaned dataframe of Obesity
df_obesity_clean.head()

Region  Gender  Year  LowerBound  UpperBound  \
1         South-East Asia    Male  2013    3.004463    4.399506   
4   Eastern Mediterranean    Male  2020   15.793777   19.245727   
11  Eastern Mediterranean    Male  2020   23.089589   33.815336   
17                 Europe  Female  2014   24.924766   31.213260   
19                 Europe  Female  2019   28.160682   35.574925   

    Mean_Estimate                    Country Age_group   CI_Width  \
1        3.682037                  Sri Lanka    adults   1.395044   
4       17.487641  Iran, Islamic Republic of    adults   3.451950   
11      28.335530                       Iraq    adults  10.725747   
17      27.948704                  Greenland    adults   6.288494   
19      31.893221                    Armenia    adults   7.414243   

   Obesity_level  
1            Low  
4            Low  
11      Moderate  
17      Moderate  
19          High

In [28]:
# Hence all requirements are completed so displaying cleaned dataframe of Obesity
df_malnutrition_clean.head()

Region  Gender  Year  LowerBound  UpperBound  Mean_Estimate  \
0                 Africa  Female  2021    3.972036    8.077021       5.795110   
4        Western Pacific    Both  2017    2.428230    3.842155       3.090652   
5                 Africa  Female  2016    6.655668   10.253121       8.371724   
7                   None    Both  2021   12.242323   14.442866      13.336353   
9  Eastern Mediterranean    Male  2014    1.253843    3.516819       2.208070   

                  Country Age_group  CI_Width Malnutrition_level  
0                   Gabon    adults  4.104985                Low  
4                Mongolia    adults  1.413925                Low  
5            Sierra Leone    adults  3.597453                Low  
7  South-East Asia Region    adults  2.200543           Moderate  
9                   Qatar    adults  2.262976                Low

In [57]:
# Checking count of rows and columns for understanding structure of data
df_obesity_clean.shape

(27720, 10)

In [58]:
# Checking count of rows and columns for understanding structure of data
df_malnutrition_clean.shape

(27720, 10)

In [59]:
# Checking if any null values present in data farme
df_obesity_clean.isnull().sum()

Region           1452
Gender              0
Year                0
LowerBound          0
UpperBound          0
Mean_Estimate       0
Country             0
Age_group           0
CI_Width            0
Obesity_level       0
dtype: int64

In [60]:
# Checking if any null values present in data farme
df_malnutrition_clean.isnull().sum()

Region                1452
Gender                   0
Year                     0
LowerBound               0
UpperBound               0
Mean_Estimate            0
Country                  0
Age_group                0
CI_Width                 0
Malnutrition_level       0
dtype: int64

There are 1452 Null values present in region and the country are not specified to locate the region.

In [62]:
# Checking if any duplicates present in data frame.
df_obesity_clean.duplicated().sum()

np.int64(0)

In [63]:
# Checking if any duplicates present in data frame.
df_malnutrition_clean.duplicated().sum()

np.int64(0)

In [66]:
# Using some visulaization techniques to explore data frame
import plotly.express as px
px.scatter(df_obesity_clean,x="Mean_Estimate",y="CI_Width",labels={"x":"Mean_Estimate","y":"CI_Width"},title = "Mean_Estimate Vs CI_Width")

In [67]:
px.bar(df_obesity_clean,x="Year",y="Mean_Estimate",hover_name = "CI_Width",color="Obesity_level" )# X axis can be a time related field or category related
# Y axis can be continuous data

From the above visualization, Observed that percentage of obesity increases throughout years passed.

The Obseity level also increased throughout years

In [70]:
px.pie(df_obesity_clean,names="Year",values="CI_Width")

From the above visualization, Observed that 2021 has higher confidence interval

In [69]:
px.histogram(df_obesity_clean,x ="CI_Width",nbins = 10)

In [ ]:
# Extarcted the data into csv format to visualize result in excel sheeet
df_obesity_clean.to_csv('Obesity.csv', index=False)
df_malnutrition_clean.to_csv('Malnutrition.csv', index=False)

In [55]:
# Code written to make connection with SQL database and current Juypiter notebook via TiDB cloud host.
!pip install mysql-connector-python # installation of SQL connector
import mysql.connector
suhail = mysql.connector.connect(host="gateway01.ap-southeast-1.prod.aws.tidbcloud.com",
                                 user="2XT5EKXhUVwb2JW.root",
                                 password="YO7YdnLJV098Hccw",
                                 port = 4000) # Establish DB connection using TiDB cloud

In [47]:
# Code written to create database and tables
connection = suhail.cursor()
connection.execute("CREATE DATABASE Nutrition_Paradox")
connection.execute("USE Nutrition_Paradox")
connection.execute("CREATE TABLE obesity (Region varchar(50), Gender varchar(10), Year int, LowerBound float, UpperBound float, Mean_Estimate float, Country varchar(50), Age_group varchar(10), CI_Width float, Obesity_level varchar(10))")
connection.execute("CREATE TABLE malnutrition (Region varchar(50), Gender varchar(10), Year int, LowerBound float, UpperBound float, Mean_Estimate float, Country varchar(50), Age_group varchar(10), CI_Width float, Malnutrition_level varchar(10))")

In [46]:
# Run this line if we receive error as database or table already exists
connection = suhail.cursor()
connection.execute("DROP DATABASE Nutrition_Paradox")

In [53]:
# Insertion of data from data frame into SQL table as per requirement.
for index, row in df_obesity_clean.iterrows():
    connection.execute('''INSERT INTO obesity (Year, Gender, Mean_Estimate, LowerBound, UpperBound,
     Age_group, Country, Region, CI_Width, Obesity_level) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)''',
                       (row['Year'], row['Gender'], row['Mean_Estimate'], row['LowerBound'], row['UpperBound'],
                        row['Age_group'], row['Country'], row['Region'], row['CI_Width'], row['Obesity_level']))

ProgrammingError: Not all parameters were used in the SQL statement

In [56]:
data = df_obesity_clean[[
    'Year', 'Gender', 'Mean_Estimate', 'LowerBound', 'UpperBound',
    'Age_group', 'Country', 'Region', 'CI_Width', 'Obesity_level'
]].values.tolist()

connection.executemany('''
    INSERT INTO obesity
    (Year, Gender, Mean_Estimate, LowerBound, UpperBound,
     Age_group, Country, Region, CI_Width, Obesity_level)
    VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
''', data)

ProgrammingError: Not all parameters were used in the SQL statement